node-graph/gstd/src/raster.rs

```diff
@@ -130,17 +130,86 @@ pub struct BlendImageNode<Second, MapFn> {
 	map_fn: MapFn,
 }
 
+fn sample_from_image(image: &Image, u: f64, v: f64) -> Color {
+	let x = ((u * image.width as f64) as u32).clamp(0, image.width);
+	let y = ((v * image.height as f64) as u32).clamp(0, image.height);
+
+	image.data[(y * image.width + x) as usize]
+}
+
+struct AxisAlignedBbox {
+	start: DVec2,
+	end: DVec2,
+}
+
+struct Bbox {
+	top_left: DVec2,
+	top_right: DVec2,
+	bottom_left: DVec2,
+	bottom_right: DVec2,
+}
+
+impl Bbox {
+	fn axis_aligned_bbox(&self) -> AxisAlignedBbox {
+		let start_x = self.top_left.x.min(self.top_right.x).min(self.bottom_left.x).min(self.bottom_right.x);
+		let start_y = self.top_left.y.min(self.top_right.y).min(self.bottom_left.y).min(self.bottom_right.y);
+		let end_x = self.top_left.x.max(self.top_right.x).max(self.bottom_left.x).max(self.bottom_right.x);
+		let end_y = self.top_left.y.max(self.top_right.y).max(self.bottom_left.y).max(self.bottom_right.y);
+
+		AxisAlignedBbox {
+			start: DVec2::new(start_x, start_y),
+			end: DVec2::new(end_x, end_y),
+		}
+	}
+}
+
+fn compute_transformed_bounding_box(image_frame: &ImageFrame) -> Bbox {
+	let top_left = DVec2::new(0.0, 0.0);
+	let top_right = DVec2::new(image_frame.image.width as f64, 0.0);
+	let bottom_left = DVec2::new(0.0, image_frame.image.height as f64);
+	let bottom_right = DVec2::new(image_frame.image.width as f64, image_frame.image.height as f64);
+	let transform = |p| image_frame.transform.transform_point2(p);
+
+	Bbox {
+		top_left: transform(top_left),
+		top_right: transform(top_right),
+		bottom_left: transform(bottom_left),
+		bottom_right: transform(bottom_right),
+	}
+}
+
 // TODO: Implement proper blending
 #[node_macro::node_fn(BlendImageNode)]
 fn blend_image<MapFn>(image: ImageFrame, second: ImageFrame, map_fn: &'any_input MapFn) -> ImageFrame
 where
 	MapFn: for<'any_input> Node<'any_input, (Color, Color), Output = Color> + 'input,
 {
-	let mut image = image;
-	for (pixel, sec_pixel) in &mut image.image.data.iter_mut().zip(second.image.data.iter()) {
-		*pixel = map_fn.eval((*pixel, *sec_pixel));
+	let (mut dst_image, src_image) = (image, second);
```

Comment:
I think I should maybe rename the arguments? I thought `second` needs to be blended on top of `image`. I think it would be more intuitive to name them `source` and `destination` from the beginning.

Reply:
Yeah, the names are terrible you can take a look at the definition of the blend function itself to verify the order of operands